cxx-qt-gen/src/syntax/fields.rs

```diff
@@ -0,0 +1,121 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use syn::{spanned::Spanned, Error, Field, Fields, FieldsNamed, Ident, Result, Type};
+
+/// In a group of [syn::Fields] extract the [syn::Ident] and [syn::Type] from
+/// any [syn::FieldsNamed] fields
+///
+/// If there are [syn::FieldsUnnamed] then an error occurs
+pub fn fields_named_to_ident_type(fields: &Fields) -> Result<Vec<(Ident, Type)>> {
```

Comment:
This should probably just be a function on Parameter and return a Vec<Parameter>.
 e.g. `Parameter::from_fields`.

Reply:
This might be used later for other things like parsing the struct with it's named fields (although we need the attributes info as well). But i can see us later having our own Vec<NamedField> thing which has the ident, type, attributes etc potentially we'll see. So any general syntax helpers that the parser (or other phases may use) i've been splitting into this syntax crate.